src/main/java/picard/nio/GoogleStorageUtils.java

```diff
@@ -0,0 +1,46 @@
+package picard.nio;
+
+import com.google.cloud.http.HttpTransportOptions;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.storage.contrib.nio.CloudStorageConfiguration;
+import com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider;
+import shaded.cloud_nio.com.google.api.gax.retrying.RetrySettings;
+import shaded.cloud_nio.org.threeten.bp.Duration;
+
+
+/**
+ * Created by farjoun on 11/13/17.
+ */
+class GoogleStorageUtils {
```

Comment:
Cannot GCS be configured in a different way? It looks wierd to me that there is support for the GCS in Picard, but not include the support by default.

It would be nice that both HTSJDK and Picard can use any `java.nio.Path` as a plugin, without extra classes supporting specific filesystems that are afterwards not included in the binary and requires special command line options...

Reply:
I agree that it would be good...but I'm not sure how it would be possible to accomplish that.

The current PR allows GCS access to (some) picard tools via GATK. The reason I didn't want to include the google-nio artifact is that it's huge...adding it more than doubles the size of the picard jar....but without the changes I have made here, it's effectively impossible to provide GCS access to picard tools....so I think that it's a good compromise, but I'm happy to hear other opinions.

Reply:
It's a good compromise, require the dependency for compiling only, but don't package it by default (i.e. it's an optional dependency).

Reply:
If the `Path` api provided a generic method for configuration of FileSystems properties, than we could use that instead.  It doesn't though, and this is the only way that we know to configure important settings in the google-nio library.  The problem is that the default settings that google chose make sense for general use, but not for jobs that last days and make millions of requests.  Since our use case is weird, we need to configure these weird settings.  I don't see any way to do that without including the library at compile time.  

It's likely that we'll want little performance tweaks to whatever nio libraries we work with.  They will work out of the box without them, like the google one would, but most will work better if we configure them specifically for our application.  

Maybe any confusion could be fixed by adding documentation here that would explain that this is for an optional dependency.